python/lsst/obs/base/ingest.py

```diff
@@ -920,6 +920,69 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
             self.log.info("Exposure %s:%s ingested successfully",
                           exposure.record.instrument, exposure.record.obs_id)
 
+        return refs, bad_files, n_exposures, n_exposures_failed, n_ingests_failed
+
+    @timeMethod
+    def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Optional[str] = None,
+            file_filter: Union[str, re.Pattern] = r"\.fit[s]?\b", group_files: bool = True):
+        """Ingest files into a Butler data repository.
+
+        This creates any new exposure or visit Dimension entries needed to
+        identify the ingested files, creates new Dataset entries in the
+        Registry and finally ingests the files themselves into the Datastore.
+        Any needed instrument, detector, and physical_filter Dimension entries
+        must exist in the Registry before `run` is called.
+
+        Parameters
+        ----------
+        files : iterable over `ButlerURI`, `str` or path-like objects
```

Comment:
Does that mean this will now work with `pathlib.Path` objects, too?  I rediscovered those in my repo setup work recently and was ever so slightly sad to discovered that either this or `ConvertRepoTask` couldn't handle them.

Reply:
ButlerURI handles path-like objects.